frontends/common/constantFolding.cpp

```diff
@@ -78,7 +78,7 @@ const IR::Expression* DoConstantFolding::getConstant(const IR::Expression* expr)
 }
 
 const IR::Node* DoConstantFolding::postorder(IR::PathExpression* e) {
-    if (refMap == nullptr)
+    if (refMap == nullptr || assignmentTarget)
```

Comment:
If you use P4WriteContext here, there's no need for all the extra cruft with assignmentTarget and this becomes a one-line change

    if (refMap == nullPtr || isWrite())

This will incidentally also not fold into an out or inout param of a method/function